### ISSUE TITLE
IOS app development which

### ISSUE THREAD

**Issue by Wangyida** [tiny-dnn/tiny-dnn#302]
@bhack @nyanp @edgarriba I add a new branch for IOS development demo together with a simple readme, please give me some advices or questions for this(I think there might be all questions for these codes). I can modify the codes with some friends in school.
BTW, I will modify the conv and deconv pipeline based on this [conv_arithmetic](https://github.com/vdumoulin/conv_arithmetic) rather than my previous one, @bhack do you think this is better? Seems a general convolution related topic.

**Comment by bhack** [comment]
For dependencies you could see the right column of [Hunter](https://github.com/ruslo/hunter/wiki/usr.variables) packages. This is related to [Polly](https://github.com/ruslo/polly) that we could evaluate for better cmake cross platform support.

**Comment by Wangyida** [comment]
@bhack Yeah these flags are useful enough in cmake style, as tiny_dnn only have headers in the core codes, I think it's unnecessary to make a ```libtiny_dnn``` library or Framework file for IOS xcodeproject. So I think if we should using cmake for IOS development demo, there might be better design all examples in tiny_dnn into libraries for Xcode development, what do you think about this aspect? So user could introduce these example libs into there IOS app on Xcode.

**Comment by ruslo** [comment]
> A lot of implementation is involved with hybrid programming with C++ and Objective-C

What is the problem exactly? They are compatible, as far as I know you can use `*.hpp` C++ header for both Objective-C++ `*.mm` and pure C++ `*.cpp` sources, in CMake code you can mix sources in one target. Also you can take a look at [djinni](https://github.com/dropbox/djinni) (I have not tried it yet and it's not in Hunter yet).

> I haven't find a proper way to introduce OpenCV and Protobuf libs/Framework into Xcode project by CMake

If you decide to use static libraries you can use Hunter to download those dependencies. Protobuf and OpenCV supported for iOS:
* https://travis-ci.org/ingenue/hunter/builds/154804299
* https://travis-ci.org/ingenue/hunter/builds/157581834

> We have also needs to support Android at #256?

Hunter do support Android too. Usually I do test stuff on OSX host, rare on Linux host.

I can say more if you ask focused question :)
